### DISCUSSION TITLE
Trigger the API in a chat buffer by `:w<CR>`

### DISCUSSION THREAD

**Discussion by mrjones2014** [olimorris/codecompanion.nvim#27] [Ideas]
Using `buftype=acwrite` allows you to take over the buffer read and write operations and handle them manually via autocmds `BufReadCmd` and `BufWriteCmd`, respectively. I have [an example in op.nvim](https://github.com/mrjones2014/op.nvim/blob/b8f271417acee8db09db6c43b1d496ecb80f7e4c/lua/op/securenotes/init.lua#L97-L108). You could use this to have `:w` in the chat buffer act the same as the default `<C-s>` keybind.

**Comment by olimorris** [comment]
That's a great suggestion!

**Reply by mrjones2014** [reply]
Actually looking at the code it looks like you've already done this, I just got confused because `ollama` was pulling the model so there was a huge delay.

It could be a good enhancement to add a loading indicator to the chat buffer or something. I can try to work on a PR for that.

**Reply by olimorris** [reply]
Ah I thought you were referencing calling the keymap mapping via `:w` which would allow a user to override that via the config.

Also, more than happy for a spinner PR!
